docs/metrics/diarization-error.md

```diff
@@ -0,0 +1,103 @@
+# Diarization Error Rate
+Diarization Error Rate (DER) is an important metric used to evaluate speaker diarization systems. It quantifies the overall performance of a speaker diarization system by measuring the ratio of the duration of errors to the total reference speech duration. DER is calculated as a score between 0 and 1, with 0 being a perfect match and 1 being a complete mismatch between candidate and reference diarization.
+
+## Implementation Details
+### Background
+To simplest way to quantify the error in a candidate diarization is to measure the duration of false alarms, missed detections, and speaker confusions. These three elementary errors form the building blocks of diarization error rate.
+
+??? example "False Alarm"
+    False alarm is duration of non-speech classified as speech — analagous to false positive diarizations. Using the following hypothetical reference and candidate diarization segments, let's calculate our false alarm duration.
+
+    ```python
+    from pyannote.core import Segment, Annotation
+
+    reference = Annotation()
+    reference[Segment(0, 10)] = 'A'
+    reference[Segment(13, 21)] = 'B'
+    reference[Segment(23, 32)] = 'A'
+    reference[Segment(32, 40)] = 'C'
+
+    hypothesis = Annotation()
+    hypothesis[Segment(2, 14)] = 'A'
+    hypothesis[Segment(14, 15)] = 'B'
+    hypothesis[Segment(15, 20)] = 'A'
+    hypothesis[Segment(23, 36)] = 'D'
+    hypothesis[Segment(36, 40)] = 'C'
+    ```
+
+    ![Visualization](../assets/images/diarization1.png)
+
+    Upon inspection, at time segments `[10, 13]` and `[23, 24]`, our candidate diarization contains speech that doesn't exist in the reference diarization. Thus, our false alarm is equal to $3 + 1 = 4$ seconds.
+
+
+??? example "Missed Detection"
+    Missed detection is the duration of speech classified as non-speech — analagous to a false negative in our diarizations. Using the previous example, let's calculate our missed detection duration.
+
+    ![Visualization](../assets/images/diarization1.png)
+
+    The model failed to detect speech at time segments `[0, 2]` and `[20, 21]`. Thus, our missed detection detection sums up to 3 seconds.
+
+??? example "Speaker Confusion"
+    Speaker confusion is the duration of speech that has been misidentified. Once again, lets use the previous example to calculate our speaker confusion duration.
+
+    ![Visualization](../assets/images/diarization1.png)
+
+    In the segment `[13, 14]`, our candidate misidentifies speaker `B` as speaker `A`. In segment `[14, 15]`, our candidate misidentifies `B` as `C`. In segment `[24, 32]`, the candidate misidentifies `A` as `C`. Finally, in segment `[36, 40]`, `C` is misidentified as `D`. In total, our speaker confusion amounts to 13 seconds.
```

Comment:
Shouldn't it be 14 seconds?

Reply:
Yup, you're right - miscalculation on my end
<img width="358" alt="image" src="https://github.com/kolenaIO/kolena/assets/31229051/38bf784f-445a-4b04-856a-3277d39e3952">